src/modules/UI/Settings/reducer.js

```diff
@@ -334,15 +340,17 @@ export const settingsLegacy = (state: SettingsState = initialState, action: Acti
     }
 
     case 'SET_TOKEN_SETTINGS': {
-      const { currencyCode } = data
+      if (!action.data) throw new Error('Invalid action')
+      const { currencyCode } = action.data
       return {
         ...state,
-        [currencyCode]: data
+        [currencyCode]: action.data
       }
     }
 
     case 'ADD_NEW_CUSTOM_TOKEN_SUCCESS': {
-      const { tokenObj, newCurrencyCode, settings } = data
+      if (!action.data) throw new Error('Invalid action')
+      const { tokenObj, newCurrencyCode, settings } = action.data
       const customTokens = settings.customTokens
```

Comment:
Do we need to check the existence of `settings` before we try to access the `settings.customTokens` property?